hs_collection_resource/utils.py

```diff
@@ -30,3 +39,112 @@ def add_or_remove_relation_metadata(add=True, target_res_obj=None, relation_type
+            # prepare csv content
+            csv_content_list = []
+            # create headers
+            csv_header_row = ['Title',
+                              'Type',
+                              'ID',
+                              'URL',
+                              'Owners',
+                              'Sharing Status'
+                              ]
+            csv_content_list.append(csv_header_row)
+            # create rows for currently contained resources
+            for res in collection_obj.resources.all():
+                csv_data_row = [res.metadata.title,
+                                res.resource_type,
+                                res.short_id,
+                                RES_LANDING_PAGE_URL_TEMPLATE.format(res.short_id),
+                                _get_owners_string(list(res.raccess.owners.all())),
+                                _get_sharing_status_string(res)
+                                ]
+                csv_content_list.append(csv_data_row)
+
+            # create rows for deleted resources
+            for deleted_res_log in collection_obj.deleted_resources:
+                csv_data_row = [deleted_res_log.resource_title,
+                                deleted_res_log.resource_type,
+                                deleted_res_log.resource_id,
+                                DELETED_RES_STRING,
+                                _get_owners_string(list(deleted_res_log.resource_owners.all()))
+                                if deleted_res_log.resource_owners.count() > 0
+                                else DELETED_RES_STRING,
+                                DELETED_RES_STRING
+                                ]
+                csv_content_list.append(csv_data_row)
+
+            # create a new csv on django server
+            tmp_dir = tempfile.mkdtemp()
+            csv_full_path = tmp_dir + "/" + csv_full_name
+            with open(csv_full_path, 'w') as csv_file_handle:
+                w = csv.writer(csv_file_handle)
+                for row in csv_content_list:
+                    w.writerow(row)
+
+            # push the new csv file to irods bag
+            files = (UploadedFile(file=open(csv_full_path, 'r'), name=csv_full_name))
+            add_resource_files(collection_obj.short_id, files)
+
+    except Exception as ex:
+        raise Exception("update_collection_list_csv error: " + ex.message)
+    finally:
+        if keep_local_csv:
+            return csv_full_path
+        else:
+            if tmp_dir is not None:
+                shutil.rmtree(tmp_dir)
+            return None
+
+
+def _get_owners_string(owners_list):
+
+    name_list = []
+    for owner in owners_list:
+        if owner.first_name:
+            name_str = "{0} {1}".format(owner.first_name, owner.last_name)
+        else:
+            name_str = owner.username
+        name_list.append(name_str)
+    if len(name_list) > 1:
+        return '"' + ', '.join(name_list) + '"'
```

Comment:
@pkdash when multiple owners exist, we use comma (,) to separate them like. the whole string will be written in to csv file as one cell. Since csv uses comma to separate cells, we need to add a double quote to wrap that string, like "Drew, Pabitra, Dave"